src/middlewared/middlewared/plugins/interface/lag.py

```diff
@@ -11,30 +11,75 @@ class Config:
     @private
     def lag_setup(self, lagg, members, disable_capabilities, parent_interfaces, sync_interface_opts):
         name = lagg['lagg_interface']['int_interface']
-        self.logger.info('Setting up {}'.format(name))
+        self.logger.info(f'Setting up {name}')
 
         try:
             iface = netif.get_interface(name)
         except KeyError:
             iface = None
         else:
-            first_port = next(iter(iface.ports))
+            first_port = next(iter(iface.ports), None)
             if first_port is None or first_port[0] != members[0]['lagg_physnic']:
-                self.logger.info('Destroying existing %s as its first port has changed', name)
+                self.logger.info(f'Destroying {name} because its first port has changed')
                 netif.destroy_interface(name)
                 iface = None
 
         if iface is None:
             netif.create_interface(name)
             iface = netif.get_interface(name)
 
+        # FIXME: this will fail on SCALE since we don't even have
+        # a "disable_capabilities" method
+        """
         if disable_capabilities:
             self.middleware.call_sync('interface.disable_capabilities', name)
+        """
 
+        info = {
+            'changed': False,
+            'protocol': None,
+            'xmit_hash_policy': None,
+            'lacpdu_rate': None,
+        }
         protocol = getattr(netif.AggregationProtocol, lagg['lagg_protocol'].upper())
         if iface.protocol != protocol:
-            self.logger.info('{}: changing protocol to {}'.format(name, protocol))
-            iface.protocol = protocol
+            info['changed'] = True
```

Comment:
Can we maybe just check `changed` at the end of this if clauses and see if any value has a value differnet then `None`, we can consider this `changed` ?